web/src/CityContentSwitcher.tsx

```diff
@@ -42,27 +44,23 @@ const SearchPage = lazyWithRetry(() => import('./routes/SearchPage'))
 const DisclaimerPage = lazyWithRetry(() => import('./routes/DisclaimerPage'))
 
 type CityContentSwitcherProps = {
-  cities: CityModel[]
   languageCode: string
 }
 
 export type CityRouteProps = {
-  cities: Array<CityModel>
-  cityModel: CityModel
-  languages: Array<LanguageModel>
-  languageModel: LanguageModel
+  city: CityModel | null
   pathname: string
   cityCode: string
   languageCode: string
 }
 
-const CityContentSwitcher = ({ cities, languageCode }: CityContentSwitcherProps): ReactElement => {
+const CityContentSwitcher = ({ languageCode }: CityContentSwitcherProps): ReactElement => {
   const cityCode = useParams().cityCode!
+  const { data: city, loading } = useLoadFromEndpoint(createCityEndpoint, cmsApiBaseUrl, { city: cityCode })
   const pathname = normalizePath(useLocation().pathname)
   const { viewportSmall } = useWindowDimensions()
-  const cityModel = cities.find(it => it.code === cityCode)
 
-  if (!cityModel) {
+  if (!city && !loading) {
```

Comment:
This is the important change that renders the routes and therefore loads the other requests without waiting for the loading of the city.